diff --git a/src/luks/tests/meson.build b/src/luks/tests/meson.build
index a1b6c1d..2f7b6b3 100644
--- a/src/luks/tests/meson.build
+++ b/src/luks/tests/meson.build
@@ -3,7 +3,7 @@ jq = find_program('jq', required: false)

 # We use cryptsetup for testing LUKS2 binding and saving the token in a
 # given token slot.
-cryptsetup = find_program('cryptsetup', required: true)
+cryptsetup = find_program('cryptsetup', required: false)

 # Use keyctl to check an existing token id can be created from
 # kernel keyring password
@@ -73,6 +73,7 @@ test('bind-luks1-avoid-luksmeta-corruption', find_program('bind-luks1-avoid-luk
 # LUKS2 tests go here, and they get included if we get support for it, based
 # on the cryptsetup version.
 # Binding LUKS2 takes longer, so timeout is increased for a few tests.
+if cryptsetup.found()
 if luksmeta_data.get('OLD_CRYPTSETUP') == '0'
   test('bind-wrong-pass-luks2', find_program('bind-wrong-pass-luks2'), env: env)
   test('bind-luks2', find_program('bind-luks2'), env: env, timeout: 60)
@@ -102,5 +103,6 @@ if luksmeta_data.get('OLD_CRYPTSETUP') == '0'
   test('backup-restore-luks2', find_program('backup-restore-luks2'), env: env, timeout: 120)
   test('pass-tang-luks2', find_program('pass-tang-luks2'), env: env, timeout: 60)
 endif
+endif

 test('unlock-arbitrary-parameter', find_program('unlock-arbitrary-parameter'), env: env)
